ats/pkb/pkb.py

```diff
@@ -0,0 +1,57 @@
+from ats.ast import nodes
+
+
+def process_follows(query, statements):
+    import json
+
+    print(json.dumps(query, indent=2))
+
+    return []
+
+
+def create_follows_dictionary(tree: nodes.ProgramNode):
+    parent_childrens_list = []
+    follows_dictionary = {}
+
+    def create_parent_childrens_list(node: nodes.ASTNode):
+        childrens = []
+        for i in range(len(node.children)):
+            create_parent_childrens_list(node.children[i])
+            childrens.append(node.children[i])
+        if len(node.children) > 0:
+            parent_childrens_list.insert(0, childrens)
+
+    create_parent_childrens_list(tree)
+
+    for i in range(len(parent_childrens_list)):
+        if len(parent_childrens_list[i]) > 1:
+            for j in range(len(parent_childrens_list[i]) - 1):
+                if (
+                    str(parent_childrens_list[i][j]).split(": ")[0] == "stmt"
+                    and str(parent_childrens_list[i][j]).split(": ")[0]
+                    == str(parent_childrens_list[i][j + 1]).split(": ")[0]
+                ):
+                    follows_dictionary[str(parent_childrens_list[i][j])] = str(
+                        parent_childrens_list[i][j + 1]
+                    )
```

Comment:
Dlaczego jako klucze macie np `stmt: assign` a jako wartosc macie np `stmt: if`. Co wam to da? Finalnie macie przeciez zwrocic numerek statementu a nie jego graficzna reprezentacje w drzewie AST